app.py

```diff
@@ -1,44 +1,62 @@
+# app-puakaldi-wrapper version 0.1.0
+# author: Angus L'Herrou
+# org: CLAMS team
+
 import json
-import os, shutil
+import os
+import shutil
 import subprocess
-from typing import Dict
+from typing import Dict, Sequence, Tuple, List, Union
+import argparse
 
 from clams import ClamsApp, Restifier
 from mmif import Mmif, View, Annotation, Document, AnnotationTypes, DocumentTypes, Text
+from lapps.discriminators import Uri
 
-KALDI_MEDIA_DIRECTORY = '/audio-in'
-KALDI_16KHZ_DIRECTORY = '/audio_in_16khz'
-KALDI_EXPERIMENT_DIR = '/kaldi/egs/american-archive-kaldi/sample_experiment'
+MEDIA_DIRECTORY = '/audio_in'
+MEDIA_DIRECTORY_16KHZ = '/audio_in_16khz'
+KALDI_AMERICAN_ARCHIVE = '/kaldi/egs/american-archive-kaldi'
+KALDI_EXPERIMENT_DIR = os.path.join(KALDI_AMERICAN_ARCHIVE, 'sample_experiment')
 KALDI_OUTPUT_DIR = os.path.join(KALDI_EXPERIMENT_DIR, 'output')
-KALDI_VERSION = 'IDK'
+TRANSCRIPT_DIR = os.path.join(MEDIA_DIRECTORY, 'transcripts')
+KALDI_VERSION = '0.1.0'
+TOKEN_PREFIX = 't'
 TEXT_DOCUMENT_PREFIX = 'td'
 TIME_FRAME_PREFIX = 'tf'
 ALIGNMENT_PREFIX = 'a'
 
 
 class Kaldi(ClamsApp):
-    def appmetadata(self):
-        metadata = {
+
+    def setupmetadata(self) -> dict:
+        return {
             "name": "Kaldi Wrapper",
             "description": "This tool wraps the Kaldi ASR tool",
             "vendor": "Team CLAMS",
             "iri": f"http://mmif.clams.ai/apps/kaldi/{KALDI_VERSION}",
```

Comment:
In addition to `APP_VERSION`, we might also need to keep the record of underlying wrapped tool using something like `WRAPPEE_VERSION` (we don't have a fixed app metadata schema yet). For that there are two versions here, one is the hipstas docker image version (`v1`) and the kaldi version. As kaldi have had a quite complicated history of versioning, the most reliable way to refer a kaldi version would be, I think, the git commit id, which in this image can be obtained via 
```
subprocess.run("git --git-dir=/kaldi/.git rev-parse HEAD".split(), stdout=subprocess.PIPE).stdout.decode('ascii')[:6]
```

I realized that there are several layers of *wrapping* in this app, and how frequent it would the case for other tools. So considering all that, we can concat both versions to use as `WRAPPEE_VERSION` or pick only one.

Reply:
My vote would be just to use the docker image version, since the Kaldi version will only change if the image version changes, right?